vpr/src/route/route_tree_timing.cpp

```diff
@@ -1010,6 +1055,21 @@ t_rt_node* prune_route_tree_recurr(t_rt_node* node, CBRR& connections_inf, bool
         }
 
     } else {
+        // If this node is:
+        //   1. Part of a non-configurable node set
+        //   2. The first node in the tree that is part of the non-configurable
+        //      node set
+        //
+        //      -- and --
+        //
+        //   3. The node set is not active
+        //
+        //  Then prune this node.
+        if (non_config_node_set_usage != nullptr && node_set != -1 && device_ctx.rr_switch_inf[node->parent_switch].configurable() && (*non_config_node_set_usage)[node_set] == 0) {
+            // This node should be pruned, re-prune edges once more.
+            return prune_route_tree_recurr(node, connections_inf, /*force_prune=*/false, non_config_node_set_usage);
```

Comment:
The condition here seems correct.

Maybe I'm missing something, but its not immediately clear to me how the call to prune_route_tree_recurr() with force_prune=false actually causes the node set to be pruned. Wouldn't it just be a no-op? The conditions which brought us to this branch of prune_route_tree() with the current node haven't changed.

Reply:
If the following is true: 
- The node set is unused (e.g. `(*non_config_node_set_usage)[node_set] == 0`)
- This particular node still had children (which is true by virtue of being in this else statement)

Then that indicates that the node set became unused during the pruning.  One or more of the children of this node will be prune if prune_route_tree_recurr is called again.

Consider:
```
1 -> 2
2 -> 3 [non-configurable]
2 -> 4 [non-configurable]
3 -> 5
4 -> 6
```

Assume that nodes 5 and 6 do *not* connect to a sink, so they will be pruned (as normal).  When `prune_route_tree_recurr` visits 2 for the first time, node 3 or 4 will remain.  This is because when `prune_route_tree_recurr` visits 3 or 4, it will *not* have visited 4 or 3 (respectively).  As a result, either node 3 or 4 will not be pruned on the first pass, because the node set usage count will be > 0.  However after `prune_route_tree_recurr` visits 2, 3 and 4, the node set usage will be 0, so everything can be pruned.

Reply:
OK that makes more sense. I think we need more comments to discuss this case, probably including your example.

Reply:
I've added this example for now.  If we can come up with a pithy'er comment, we can shorted it.